app.py

```diff
@@ -104,6 +105,40 @@ def load_adventure_assignments_per_level(lang, level):
 
 
 app = Flask(__name__, static_url_path='')
+
+# set the security stuff https://github.com/GoogleCloudPlatform/flask-talisman
+content_security_policy = {
+    'default-src': [
+        '\'self\'',
+        '*.cloudfront.net',
+        'youtube.com',
+    ],
+    'font-src': [
+        'fonts.googleapis.com',
+        'fonts.gstatic.com',
+    ],
+    'frame-src': [
+        'www.youtube.com',
+    ],
+    'img-src': [
+        '\'self\'',
+        'data:',
+    ],
+    'script-src': [
+        '\'self\'',
+        's3.amazonaws.com',
+        '\'unsafe-inline\'',
+        '\'unsafe-eval\'',
```

Comment:
these two are generally considered bad practice, and it might be better to try and get rid of the inline script stuff before doing this talisman addition anyway. On the other hand, I could easily add some TODO comments or something to the effect of "these two lines probably shouldn't be here, and we've created issues #XXX and #YYY to address them".

Reply:
You've probably added the `unsafe-eval` because you've determined that it is necessary. The only place I can imagine this is coming from is probably going to be `skulpt`, which is not in power to change. So if that library needs `eval()`, then we need `eval()` and that's not likely to change.

As for `unsafe-inline`... we technically could work those, but they're just so gosh-darn convenient :).

Reply:
in investigating, it is indeed skulpt (which I hadn't appreciated was giving us python parsing capability in the browser) that requires that `unsafe-eval`. In terms of the `unsafe-inline`, that is also very convenient, and I think I'm going to recommend we keep it, to be elaborated in a comment below.

Reply:
Ack.